### PR TITLE
Add jitter for exponential backoff

### PR DESCRIPTION
**Pull request by timusg**
It is extension to the [Pr](https://github.com/gojektech/heimdall/pull/22) raised by @pravj and adds the jitter for the exponential backoff

### CODE REVIEW

**Comment by rShetty** [comment]
@timusg We need to update the docs too

File: backoff.go [hunk 1]
```diff
@@ -49,5 +52,5 @@ func (eb *exponentialBackoff) Next(retry int) time.Duration {
 		return 0 * time.Millisecond
 	}
 
-	return time.Duration(math.Min(eb.initialTimeout+math.Pow(eb.exponentFactor, float64(retry)), eb.maxTimeout)) * time.Millisecond
+	return time.Duration(math.Min(eb.initialTimeout+math.Pow(eb.exponentFactor, float64(retry))+float64(rand.Int63n(eb.maximumJitterInterval)), eb.maxTimeout)) * time.Millisecond
```
Comment (hogaur) [inline backoff.go]: I think it makes sense to add jitter to the result of `Math.min`

File: backoff_test.go [hunk 2]
```diff
@@ -9,28 +9,28 @@ import (
...
-	exponentialBackoff := NewExponentialBackoff(2*time.Millisecond, 10*time.Millisecond, 2.0)
+	exponentialBackoff := NewExponentialBackoff(2*time.Millisecond, 10*time.Millisecond, 2.0, 1)
 
 	assert.Equal(t, 4*time.Millisecond, exponentialBackoff.Next(1))
 }
 
 func TestExponentialBackoffMaxTimeoutCrossed(t *testing.T) {
 
-	exponentialBackoff := NewExponentialBackoff(2*time.Millisecond, 9*time.Millisecond, 2.0)
+	exponentialBackoff := NewExponentialBackoff(2*time.Millisecond, 9*time.Millisecond, 2.0, 1)
```
Comment (hogaur) [inline backoff_test.go]: How did the tests not change ?

File: backoff.go [hunk 3]
```diff
@@ -29,17 +30,19 @@ func (cb *constantBackoff) Next(retry int) time.Duration {
...
-	maxTimeout     float64
+	exponentFactor        float64
+	initialTimeout        float64
+	maxTimeout            float64
+	maximumJitterInterval int64
 }
 
 // NewExponentialBackoff returns an instance of ExponentialBackoff
-func NewExponentialBackoff(initialTimeout, maxTimeout time.Duration, exponentFactor float64) Backoff {
+func NewExponentialBackoff(initialTimeout, maxTimeout time.Duration, exponentFactor float64, maximumJitterInterval int64) Backoff {
```
Comment (rShetty) [inline backoff.go]: can we make `maximumJitterInterval` `time.Duration` plz ?

**Comment by rShetty** [comment]
LGTM, thanks @timusg for this PR. Merging

**Pull request merged**